torchmetrics/metric.py

```diff
@@ -259,9 +259,9 @@ def reset(self):
         for attr, default in self._defaults.items():
             current_val = getattr(self, attr)
             if isinstance(default, Tensor):
-                setattr(self, attr, deepcopy(default).to(current_val.device))
+                setattr(self, attr, default.detach().clone().to(current_val.device))
             else:
-                setattr(self, attr, deepcopy(default))
+                setattr(self, attr, [])
```

Comment:
we do not need default anymore?

Reply:
since the default value of metric states can either be tensors or an empty list, in this case it always be an empty list